src/tests/data/restore.test.ts

```diff
@@ -0,0 +1,703 @@
+import { ImportedDataState } from "../../data/types";
+import { NormalizedZoomValue } from "../../types";
+import {
+  FONT_FAMILY,
+  DEFAULT_FONT_FAMILY,
+  DEFAULT_TEXT_ALIGN,
+  DEFAULT_VERTICAL_ALIGN,
+} from "../../constants";
+
+const mockSizeHelper = jest.spyOn(sizeHelpers, "isInvisiblySmallElement");
+
+beforeEach(() => {
+  mockSizeHelper.mockReset();
+});
+
+describe("restoreElements", () => {
+  it("with null array of elements", () => {
+    expect(restore.restoreElements(null)).toStrictEqual([]);
+  });
+
+  it("should not call isInvisiblySmallElement when input element is a selection element", () => {
+    const selectionEl = { type: "selection" } as ExcalidrawElement;
+    const restoreElements = restore.restoreElements([selectionEl]);
+    expect(restoreElements.length).toBe(0);
+    expect(sizeHelpers.isInvisiblySmallElement).toBeCalledTimes(0);
+  });
+
+  it("when input element type is not supported", () => {
+    const dummyNotSupportedElement = API.createElement({ type: "text" });
+
+    Object.defineProperty(dummyNotSupportedElement, "type", {
+      get: jest.fn(() => "not supported element type"),
+    });
+
+    expect(restore.restoreElements([dummyNotSupportedElement]).length).toBe(0);
+  });
+
+  it("when isInvisiblySmallElement is true", () => {
+    const rectElement = API.createElement({ type: "rectangle" });
+    mockSizeHelper.mockImplementation(() => true);
+
+    expect(restore.restoreElements([rectElement]).length).toBe(0);
+  });
+
+  it("with text element type", () => {
+    const textElement = API.createElement({
+      type: "text",
+      fontSize: 14,
+      fontFamily: 1,
+      text: "text",
+      textAlign: "center",
+      verticalAlign: "middle",
+    });
+
+    const expectedTextElement = {
+      type: textElement.type,
+      fontSize: textElement.fontSize,
+      fontFamily: textElement.fontFamily,
+      text: textElement.text,
+      baseline: textElement.baseline,
+      textAlign: textElement.textAlign,
+      verticalAlign: textElement.verticalAlign,
+    };
+
+    const restoredText = restore.restoreElements([
+      textElement,
+    ])[0] as ExcalidrawTextElement;
+
+    const gotRestoredTextElement = {
+      type: restoredText.type,
+      fontSize: restoredText.fontSize,
+      fontFamily: restoredText.fontFamily,
+      text: restoredText.text,
+      baseline: restoredText.baseline,
+      textAlign: restoredText.textAlign,
+      verticalAlign: restoredText.verticalAlign,
+    };
+
+    expect(gotRestoredTextElement).toMatchObject(expectedTextElement);
+  });
+
+  it("when text element has null text", () => {
+    const textElement = API.createElement({ type: "text" });
+
+    Object.defineProperty(textElement, "text", {
+      get: jest.fn(() => null),
+    });
+
+    const restoredText = restore.restoreElements([
+      textElement,
+    ])[0] as ExcalidrawTextElement;
+    expect(restoredText.text).toBe("");
+  });
+
+  it("when text element has undefined alignment", () => {
```

Comment:
instead of having separate assertions for all conditons, lets have two snapshots
1. One where you pass value for each attribute of text element which you are already doing above
1. One as suggested above where we can remove the fields which are set to default values

Reply:
Hi @ad1992 ! Thanks for your suggestion, indeed we need to snapshot the tests! I was wondering if we can have 3 tests for the text elements:
```javascript
// test 01
// passing value for each attribute
const textElement = API.createElement({
type:  "text",
fontSize:  14,
fontFamily:  1,
text:  "text",
textAlign:  "center",
verticalAlign:  "middle"
});
expect(restoredText).toMatchSnapshot({seed: expect.any(Number)});
```

```javascript
// test 02
// font property, text null, alignment undefined
const textElement = API.createElement({
type:  "text",
text: null,
font: "10 Cascadia",
textAlign: undefined,
verticalAlign: undefined
});
expect(restoredText).toMatchSnapshot({seed: expect.any(Number)});
```

```javascript
// test 03
// font property with unknown font family, text null, alignment undefined
const textElement = API.createElement({
type:  "text",
text: null,
font: "10 unknown",
textAlign: undefined,
verticalAlign: undefined
});
expect(restoredText).toMatchSnapshot({seed: expect.any(Number)});
```

What do you think?

Reply:
in third case only font family is unknown so lets merge it with the second case?

Reply:
sounds good to me!